### PR TITLE
(#1786126) fgets line split 7.5.z

### PR DESCRIPTION
**Pull request by jsynacek**

### CODE REVIEW

**Comment by jsynacek** [comment]
Travis CI failure is unrelated to the patch.

File: src/core/manager.c [hunk 1]
```diff
@@ -2519,21 +2520,19 @@ int manager_deserialize(Manager *m, FILE *f, FDSet *fds) {
...
-                        if (feof(f))
-                                r = 0;
-                        else
-                                r = -errno;
 
-                        goto finish;
-                }
+                r = read_line(f, LONG_LINE_MAX, &line);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to read serialization line: %m");
```
Comment (dtardon) [inline src/core/manager.c]: Eh. We screwed this up in the original backport. If this `return` ever happens, `m->n_reloading` will remain >0, with unclear consequences (but I doubt they will be positive). The `return` needs to be changed to `goto finish`.

File: src/core/manager.c [hunk 2]
```diff
@@ -2668,22 +2667,17 @@ int manager_deserialize(Manager *m, FILE *f, FDSet *fds) {
...
-                        else
-                                r = -errno;
-
-                        goto finish;
-                }
-
-                char_array_0(name);
+                r = read_line(f, LONG_LINE_MAX, &line);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to read serialization line: %m");
```
Comment (dtardon) [inline src/core/manager.c]: Ditto.

**Review by dtardon — CHANGES_REQUESTED** [review]

**Comment by jsynacek** [comment]
v2.

**Review by dtardon — APPROVED** [review]
LGTM

**Pull request merged**